### PR TITLE
Update download link and refactor the documentation directory

### PR DESCRIPTION
**Pull request by FeynmanZhou**
Fix https://github.com/notaryproject/notation/issues/307

### CODE REVIEW

**Comment by FeynmanZhou** [comment]
cc @notaryproject/notation-code-reviewers

File: README.md [hunk 1]
```diff
@@ -41,18 +41,29 @@ Notation is a project to add signatures as standard items in the registry ecosys
...
 
-For more detailed samples, see [hello-signing](docs/hello-signing.md)
 
-## Core Documents
+## Documents
 
-- [Governance for Notation](https://github.com/notaryproject/notary/blob/master/GOVERNANCE.md)
-- [Maintainers and reviewers list](https://github.com/notaryproject/notary/blob/master/MAINTAINERS)
+- [Hello World for Notation: Local signing and verification](docs/hello-signing.md)
+- [Build, sign, and verify container images using Notary and Azure Key Vault](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-tutorial-sign-build-push)
```
Comment (dtzar) [inline README.md:48]: Should not have en-us in url.

File: README.md [hunk 2]
```diff
@@ -41,18 +41,29 @@ Notation is a project to add signatures as standard items in the registry ecosys
...
+- Build, sign, and verify container images using Notary and AWS KMS (coming soon)
```
Comment (dtzar) [inline README.md:49]: Leave these two off until we have them.

**Review by dtzar — APPROVED** [review]

**Review by SteveLasker — APPROVED** [review]
LGTM

**Review by dtzar — APPROVED** [review]

**Comment by 1235467Ad** [comment]
<Summertime>And the Livings easy Valleys On The Microphone For Ross M.G.

**Pull request merged**